agents/ProductAgent.java

```diff
@@ -25,7 +25,7 @@ private void insertProduct(int iterationScopeId) {
         ).insert(
                 Graql.var("product")
                         .isa("product")
-                        .has("product-barcode", Double.valueOf(hashcode(iterationScopeId)))
+                        .has("product-barcode", Double.valueOf(uniqueID(iterationScopeId)))
```

Comment:
Why is `product-barcode` a double?

Reply:
This is solely because we need coverage of all datatypes as keys! If I encounter a better example as I go then I'll put it there